### ISSUE TITLE
Dialog footer and title width

### ISSUE THREAD

**Issue by hoseyjoe** [haxeui/haxeui-core#456]
If Dialog base
```
 public override function validateComponentLayout():Bool {
        var b = super.validateComponentLayout();
        dialogTitle.width = this.layout.innerWidth;
        if (_autoSizeDialog == false) {
            var offset = this.layout.paddingLeft + this.layout.paddingRight;
            if (dialogContent.width != this.width - offset) {
                dialogContent.width = this.width - offset;
            }
            if (dialogFooterContainer.width != this.width - offset) {
                dialogFooterContainer.width = this.width - offset;
            }
        }
        
        return b;
    }
```
Will display dialog like
![image](https://user-images.githubusercontent.com/2074385/172062901-89882e9b-ea96-400d-92ca-894d9fca696d.png)

Adjusting validateComponentLayout to
```
public override function validateComponentLayout():Bool {
        var b = super.validateComponentLayout();
        dialogTitle.width = this.layout.innerWidth;
        if (_autoSizeDialog == false) {
            var offset = this.layout.paddingLeft + this.layout.paddingRight;
            dialogTitle.width = this.layout.innerWidth - offset;
            if (dialogContent.width != this.width - offset) {
                dialogContent.width = this.width - offset;
            }
            if (dialogFooterContainer.width != this.layout.innerWidth - offset) {
                dialogFooterContainer.width = this.layout.innerWidth - offset;
            }
        }
        
        return b;
    }
```
Fixes the issue
![image](https://user-images.githubusercontent.com/2074385/172063023-69c516f9-41b2-494d-a080-c9790ce17598.png)

I dont think that is the right fix? Let me know if you need a sample project. Kinda between things so I dont have time while I maing ticket

**Comment by ianharrigan** [comment]
The buttons/title here look tiny, is that intentional?

**Comment by hoseyjoe** [comment]
It was default, I thought.  I'll check. Maybe that is causing the issue

**Comment by ianharrigan** [comment]
Certainly looks like padding (or something) is massively smaller, that said, i dont think it should behave the way it does anyway - do you have the test app to play with?

**Comment by hoseyjoe** [comment]
In truth I forgot where this came from.  I'm digging and will make test app

**Comment by hoseyjoe** [comment]
So I found the problem???

There is a table with custom itemrenderer called MyMenuItem and has a MyMenuItem.xml. In it is 
```
<style>
        .titleWPad{
            padding-left:10px;
        }
        MyMenuItem{
            padding:1px;
        }        
    </style>
```
The MyMenuItem renderer has a button that opens the dialog. Removing MyMenuItem from css results in and clicking the button results in:  
![image](https://user-images.githubusercontent.com/2074385/174397950-d586cff7-cab1-48c6-9f9b-340470e82057.png)

Im not sure how MyMenuItem css is screwing up the dialog but removing it fixes it

**Comment by ianharrigan** [comment]
Interesting, i _thought_ i removed "node matching" - though im not sure i did, but if i did (or didnt) its doing _something_... its almost like its matching _everything_

**Comment by ianharrigan** [comment]
its defo matching everything: http://haxeui.org/builder/?oopviy

That is totally wrong - nice find!

**Comment by ianharrigan** [comment]
OK, so the "node based css matching everything" bug is gone - in fact, ive reintroduced node based css matching, and improved the performance (which is why i removed it in the first place). However, i still think this issue should be left open as the dialog should still look right, even if you remove all the padding.

**Comment by hoseyjoe** [comment]
This is working as expected. Close on your dime

**Comment by ianharrigan** [comment]
Yeah, im going to leave this open as the dialog looking weird when padding=0 is still there.
